src/payment/models/pricing/pricing.controller.ts

```diff
@@ -0,0 +1,23 @@
+import { Controller, UseGuards, Body, Get } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { ApiBearerAuth, ApiExcludeEndpoint, ApiTags } from '@nestjs/swagger';
+import { RoleGuard } from 'src/shared/auth/role.guard';
+import { UserRole } from 'src/shared/auth/user-role.enum';
+import { PriceRequest, PriceResult } from './interfaces';
+import { PricingService } from './services/pricing.service';
+
+@ApiTags('pricing')
+@Controller('pricing')
+export class PricingController {
+  constructor(private readonly pricingService: PricingService) {}
+
+  @Get('price')
+  @ApiBearerAuth()
+  @ApiExcludeEndpoint()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.ADMIN))
+  async getPrice(@Body() dto: PriceRequest): Promise<PriceResult> {
+    if (process.env.ENVIRONMENT === 'test') {
```

Comment:
Environment `test` does not exist. It's `loc`, `dev` or `prd`.

Reply:
it will exist :) at some point. This is aimed to restrict using this APIs for anything else other then integration tests